exotica_core/src/kinematic_tree.cpp

```diff
@@ -1023,6 +1023,33 @@ std::vector<std::string> KinematicTree::GetKinematicChain(const std::string& beg
     return chain;
 }
 
+std::vector<std::string> KinematicTree::GetKinematicChainLinks(const std::string& begin, const std::string& end)
+{
+    // check existence of requested links
+    for (const std::string& l : {begin, end})
+    {
+        if (!tree_map_.count(l))
+        {
+            ThrowPretty("Link '" + l + "' does not exist.");
+        }
+    }
+
+    // get chain in reverse order, end...begin
+    std::vector<std::string> chain;
+    for (std::shared_ptr<const KinematicElement> l = tree_map_.at(end).lock(); l->segment.getName() != begin; l = l->parent.lock())
```

Comment:
Is this `const` required or beneficial?

Reply:
It's not required but it prevents accidental changes to the link. Alternative for loops, e.g. `for(const KinematicElement & link : links) { ... }`, use to operate on const references too.